src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -793,8 +793,86 @@ internal List<INodeLibraryDependencyInfo> NodeLocalDefinitions
             }
         }
 
+        internal List<INodeLibraryDependencyInfo> ExternalFiles
+        {
+            get
+            {
+                var externalFiles = new Dictionary<object, DependencyInfo>();
+
+                foreach (var node in Nodes)
+                {
+                    if (externalFilesDictionary.TryGetValue(node.GUID, out var dependencyInfo)) 
+                    {
+                        externalFiles[dependencyInfo.Name] = dependencyInfo;
+                    }
+
+                    if (node is CodeBlockNodeModel codeBlockNode)
+                    {
+                        var lines = codeBlockNode.Code.Split('\n');
+
+                        foreach (var line in lines)
+                        {
+                            //ignore the escape chars.
+                            string lineWithoutEscapeCharacters = line.Replace("\\\\", "\\").Replace("\"", "");
+
+                            if (lineWithoutEscapeCharacters.Length > 1 && File.Exists(lineWithoutEscapeCharacters.Substring(0, lineWithoutEscapeCharacters.Length - 1)))
+                            {
+                                var path = lineWithoutEscapeCharacters.Substring(0, lineWithoutEscapeCharacters.Length - 1);
+                                var externalFileName = Path.GetFileName(path);
+
+                                if (!externalFiles.ContainsKey(externalFileName))
+                                {
+                                    externalFiles[externalFileName] = new DependencyInfo(externalFileName, path);
+                                }
+
+                                externalFiles[externalFileName].AddDependent(node.GUID);
+                                externalFiles[externalFileName].ReferenceType = ReferenceType.External;
+                            }
+                        }
+                    }
+
+                    if(node.CachedValue != null)
```

Comment:
Does this cover nodes like string node, file path node etc.?

Reply:
Yes, it covers these nodes.